File: src/version/tests/version_bindings.py
#!/usr/bin/env python3
import unittest
from common.variables import variable_names_and_objects
import version


class VersionInfo(unittest.TestCase):

    def test_basic_version_info(self):
        for variable, get_string_function in variable_names_and_objects(version.repo_name, version.repo_author, version.repo_version, version.repo_email):
            output = get_string_function()
            self.assertIsInstance(output, str, msg=f"The {variable = } with {output = } is not the right type.")
            self.assertTrue(output, msg=f"The {variable = } with {output = } is meant to be non-trivial.")


if __name__ == '__main__':
    unittest.main()

File: src/pyarv/version/version_bindings.c
#include "version_bindings.h"
#include "version/version.h"

PyObject *repo_name_(PyObject *self, PyObject *args, PyObject *kwargs)
{
    return PyUnicode_FromString(repo_name());
}

PyObject *repo_version_(PyObject *self, PyObject *args, PyObject *kwargs)
{
    return PyUnicode_FromString(repo_version());
}

PyObject *repo_author_(PyObject *self, PyObject *args, PyObject *kwargs)
{
    return PyUnicode_FromString(repo_author());
}

PyObject *repo_email_(PyObject *self, PyObject *args, PyObject *kwargs)
{
    return PyUnicode_FromString(repo_email());
}
